src/DynamoCoreWpf/UI/GuidedTour/HostInfo.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Windows;
+using System.Windows.Controls.Primitives;
+
+namespace Dynamo.Wpf.UI.GuidedTour
+{
+    /// <summary>
+    /// This class will contain info related to the popup location and the host control in which the popup will appear over
+    /// </summary>
+    public class HostInfo
+    {
+        private string name;
+        private string hostClass;
+        private UIElement hostUIElement;
+        public PlacementMode popupPlacement;
+        private double verticalPopupOffSet;
+        private double horizontalPopupOffSet;
+
+        /// <summary>
+        /// Host Name, this property will contain the name of the host control located in the TreeView
+        /// </summary>
+        public string Name 
+        {
+            get
+            {
+                return name;
+            }
+            set
+            {
+                name = value;
+            }
+        }
+
+        /// <summary>
+        /// Host Class, this property will contain the class of the host control located in the TreeView
+        /// </summary>
+        public string HostClass
+        {
+            get
+            {
+                return hostClass;
+            }
+            set
+            {
+                hostClass = value;
+            }
+        }
+
+        /// <summary>
+        /// This property will hold the UI Element over which the popup will be shown (it should be any item in the TreeView) 
+        /// </summary>
+        public UIElement HostUIElement
+        {
+            get
+            {
+                return hostUIElement;
+            }
+            set
+            {
+                hostUIElement = value;
+            }
+        }
+
+        /// <summary>
+        /// This property will hold the placement location of the popup, for now we are just using Right, Left, Top and Bottom
+        /// </summary>
+        public PlacementMode PopupPlacement
```

Comment:
If it is meant to be public, maybe we can just auto property?

Reply:
@QilongTang changed to be a auto property